### DISCUSSION TITLE
Are there some papers that I should read so I can implement something similar to rhombus' macro system?

### DISCUSSION THREAD

**Discussion by glyh** [racket/rhombus#334] [Q&A] [answered]
As title, thanks :)

**Comment by mflatt** [comment, marked as answer]
The main existing papers are

https://doi.org/10.1145/2914770.2837620 - syntax-object representation

https://doi.org/10.1145/2371401.2371420 - enforestation

https://doi.org/10.1145/3093333.3009886 - static information

https://doi.org/10.1145/3428297 - similar to space-specific macros

Those build on Racket, Scheme, and Lisp foundations, for which there are more papers, but the above links are for the most immediate precedents. 

We have a Rhombus summary paper in submission, but nothing Rhombus-specific published, yet.
